src/operator/tensor/broadcast_reduce-inl.cuh

```diff
@@ -602,6 +602,11 @@ void Reduce(Stream<gpu> *s, const TBlob& small, const OpReqType req,
   ReduceImpl<Reducer, ndim, DType, OP>(stream, small, req, big, workspace, config);
 }
 
+template <typename Reducer, int ndim, typename DType, typename OP>
+void ReduceWithExtraMem(Stream<cpu>* s, const TBlob& small, const OpReqType req,
+                        const Tensor<cpu, 1, char>& workspace, const TBlob& big) {};
```

Comment:
Empty implementation?

Reply:
`ReduceWithExtraMem` is only used for  the cpu implementation and prevents the build from failing.

Reply:
broadcast_reduce-inl.h includes the code in broadcast_reduce-inl.cuh or some part of code (including ReduceWithExtraMem) in broadcast_reduce-inl.h based on if __CUDACC__ is defined. https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/broadcast_reduce-inl.h#L171. This causes build to fail when omitting ReduceWithExtraMem in broadcast_reduce-inl.cuh.